src/test/scala/extraction/DeclaredImplicitsTest.scala

```diff
@@ -0,0 +1,112 @@
+package extraction
+
+import framework.SemanticdbTest
+
+class DeclaredImplicitsTest extends SemanticdbTest {
+  checkExtraction(
+    "Basic information about declared implicits",
+    """
+      |package dI
+      |object basicInfo {
+      | implicit def m(a: String): String = "3"
+      |}
+    """.trim.stripMargin, { res =>
```

Comment:
It becomes even nicer here:

```scala
  checkExtraction(
    "Two declared implicits with identical code have different ids",
    """
      |package dI
      |object m1 {
      | implicit val msgDeclaration: String = "World"
      |}
      |object m2 {
      | implicit val msgDeclaration: String = "World"
      |}
    """.trim.stripMargin, { res =>
      res.normalizedImplicits should contain only (
        DeclaredImplicit(
          fqn = "_root_.dI.m1.msgDeclaration.",
          clazz = "...",
          typee = "...",
          kind = "val",
          nargs = "0
        ),
        DeclaredImplicit(
          fqn = "...",
          clazz = "...",
          typee = "...",
          kind = "val",
          nargs = "0
        )
      )

      res.funs shouldBe empty
      res.links shouldBe empty
      res.params shouldBe empty
  })
```

The point is that it is dead-simple so see what is going on